elf/segmentation/stitching.py

```diff
@@ -175,4 +185,127 @@ def _compute_overlaps(block_id):
 
     if return_before_stitching:
         return seg_stitched, seg
+
+    return seg_stitched
+
+
+def stitch_tiled_segmentation(
+    segmentation: np.ndarray,
+    tile_shape: Tuple[int, int],
+    overlap: int = 1,
+    n_threads: Optional[int] = None,
+    verbose: bool = True,
+) -> np.ndarray:
+    """Functionality for stitching segmentations tile-wise based on overlap.
+
+    Args:
+        segmentation: The input segmentation.
+        tile_shape: The shape of inidividual tiles.
+        overlap: The overlap of tiles.
+            It is responsible to compute the edge nodes for the desired overlap region.
+        n_threads: The number of threads used for parallelized operations.
+            Set to the number of cores by default.
+        verbose: Whether to print the progress bars.
+
+    Returns:
+        The stitched segmentation with merged labels.
+    """
+    shape = segmentation.shape
+    ndim = len(shape)
+    blocking = nt.blocking([0] * ndim, shape, tile_shape)
+    n_blocks = blocking.numberOfBlocks
+
+    block_segs = []
+
+    # Get the tiles from the segmentation of shape: 'tile_shape'.
+    def _fetch_tiles(block_id):
+        block = blocking.getBlock(block_id)
+        bb = tuple(slice(beg, end) for beg, end in zip(block.begin, block.end))
+        block_seg = segmentation[bb]
+        block_segs.append(block_seg)
+
+    n_threads = multiprocessing.cpu_count() if n_threads is None else n_threads
+    with futures.ThreadPoolExecutor(n_threads) as tp:
```

Comment:
This is premature optimization, you don't need to wrap it in a Threadpool as this is just accessing data, which is very fast.

Reply:
Updated it to extract the data block wise. Thanks!